src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -367,10 +373,14 @@ private void SetCapacity(int capacity)
         // Increments the index wrapping it if necessary.
         private void MoveNext(ref int index)
         {
-            // It is tempting to use the remainder operator here but it is actually much slower 
-            // than a simple comparison and a rarely taken branch.   
-            int tmp = index + 1;
-            index = (tmp == _array.Length) ? 0 : tmp;
+            // It is tempting to use the remainder operator here but it is actually much slower
+            // than a simple comparison and a rarely taken branch.
+            // JIT produces better code than with ternary operator ?:
```

Comment:
What's the change here?

Reply:
e.g. is keeping the `tmp` var better (does it push to register vs memory location on `index`)

Reply:
The `if` produces better code than the ternary.

The point with the register I have to validate.

Reply:
The _tmp-Variant_ uses registers, whilst _index_ pushes to memory. I'll update the PR, though the effect might be minimal, but it's still better :wink:

For reference:

### Original
```asm
000007fe`7b6ec5f0 488d4110        lea     rax,[rcx+10h]
000007fe`7b6ec5f4 8b10            mov     edx,dword ptr [rax]
000007fe`7b6ec5f6 ffc2            inc     edx
000007fe`7b6ec5f8 488b4908        mov     rcx,qword ptr [rcx+8]
000007fe`7b6ec5fc 395108          cmp     dword ptr [rcx+8],edx
000007fe`7b6ec5ff 7402            je      000007fe`7b6ec603
000007fe`7b6ec601 eb02            jmp     000007fe`7b6ec605
000007fe`7b6ec603 33d2            xor     edx,edx
000007fe`7b6ec605 8910            mov     dword ptr [rax],edx
000007fe`7b6ec607 c3              ret
```
### Index only
```asm
000007fe`7b71c5f0 488d4110        lea     rax,[rcx+10h]
000007fe`7b71c5f4 ff00            inc     dword ptr [rax]           ; doesn't use register
000007fe`7b71c5f6 8b10            mov     edx,dword ptr [rax]
000007fe`7b71c5f8 488b4908        mov     rcx,qword ptr [rcx+8]
000007fe`7b71c5fc 3b5108          cmp     edx,dword ptr [rcx+8]
000007fe`7b71c5ff 7504            jne     000007fe`7b71c605
000007fe`7b71c601 33d2            xor     edx,edx
000007fe`7b71c603 8910            mov     dword ptr [rax],edx
000007fe`7b71c605 c3              ret
```
### tmp-Variable
```asm
000007fe`7b71c5f0 488d4110        lea     rax,[rcx+10h]
000007fe`7b71c5f4 8b10            mov     edx,dword ptr [rax]
000007fe`7b71c5f6 ffc2            inc     edx                       ; uses register
000007fe`7b71c5f8 488b4908        mov     rcx,qword ptr [rcx+8]
000007fe`7b71c5fc 395108          cmp     dword ptr [rcx+8],edx
000007fe`7b71c5ff 7502            jne     000007fe`7b71c603
000007fe`7b71c601 33d2            xor     edx,edx
000007fe`7b71c603 8910            mov     dword ptr [rax],edx
000007fe`7b71c605 c3              ret
```